From: Tom Gundersen <teg@jklm.no>
Date: Wed, 29 Jul 2015 01:41:24 +0200
Subject: networkd-wait-online: only consider interfaces given on the
 commandline

If some interfaces are given on the commandline, ignore all others.
---
 man/systemd-networkd-wait-online.service.xml | 3 ++-
 src/network/networkd-wait-online-manager.c   | 6 ++++++
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/man/systemd-networkd-wait-online.service.xml b/man/systemd-networkd-wait-online.service.xml
index f53b337..bcc5776 100644
--- a/man/systemd-networkd-wait-online.service.xml
+++ b/man/systemd-networkd-wait-online.service.xml
@@ -80,7 +80,8 @@
         several interfaces which will be configured, but a particular
         one is necessary to access some network resources. This option
         may be used more than once to wait for multiple network
-        interfaces.</para></listitem>
+        interfaces. When used, all other interfaces are ignored.
+        </para></listitem>
       </varlistentry>
       <varlistentry>
         <term><option>--ignore=</option></term>
diff --git a/src/network/networkd-wait-online-manager.c b/src/network/networkd-wait-online-manager.c
index 1fc724f..ec44893 100644
--- a/src/network/networkd-wait-online-manager.c
+++ b/src/network/networkd-wait-online-manager.c
@@ -38,9 +38,15 @@ bool manager_ignore_link(Manager *m, Link *link) {
         assert(m);
         assert(link);
 
+        /* always ignore the loopback interface */
         if (link->flags & IFF_LOOPBACK)
                 return true;
 
+        /* if interfaces are given on the commandlin, ignore all others */
+        if (m->interfaces && !strv_contains(m->interfaces, link->ifname))
+                return true;
+
+        /* ignore interfaces we explicitly are asked to ignore */
         STRV_FOREACH(ignore, m->ignore)
                 if (fnmatch(*ignore, link->ifname, 0) == 0)
                         return true;
